Indotalent/Applications/LogAnalytics/LogAnalyticService.cs

```diff
@@ -1,26 +1,68 @@
-﻿using System.Security.Claims;
-
+﻿using AutoMapper;
 using DeviceDetectorNET;
-
 using Indotalent.Data;
+using Indotalent.DTOs;
 using Indotalent.Infrastructures.Repositories;
 using Indotalent.Models.Entities;
-
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
 using UAParser;
 
 namespace Indotalent.Applications.LogAnalytics
 {
     public class LogAnalyticService : Repository<LogAnalytic>
     {
+        private readonly IMapper _mapper;
+
         public LogAnalyticService(
             ApplicationDbContext context,
             IHttpContextAccessor httpContextAccessor,
-            IAuditColumnTransformer auditColumnTransformer) :
+            IAuditColumnTransformer auditColumnTransformer,
+            IMapper mapper) :
                 base(
                     context,
                     httpContextAccessor,
                     auditColumnTransformer)
         {
+            _mapper = mapper;
+        }
+
+        public IQueryable<LogAnalyticDto> GetAllDtos()
+        {
+            return _mapper.ProjectTo<LogAnalyticDto>(_context.Set<LogAnalytic>().AsQueryable());
+        }
+
+        public async Task<LogAnalyticDto?> GetDtoByIdAsync(int id)
+        {
+            var logAnalytic = await _context.Set<LogAnalytic>().FindAsync(id);

```

Comment:
Instead of `_context.Set<LogAnalytic>()` you should use `_context.LogAnalytic` as in line 101